File: src/lib/types/ui/like/trackLikedView.ts
export type TrackLikedView = {
  id: number;                 // Correspond au `pub id: i64` de Rust
  liked_at: string;           // Serde transforme DateTime<Utc> en string (format ISO 8601)
  path: string;
  library_cache_id: number | null;  // Option<i64> devient number | null
  
  // --- Les champs fusionnés de la bibliothèque (JOIN) ---
  title: string | null;
  artist: string | null;
  album: string | null;
  duration: number | null;
  thumbnail_path: string | null;
  bits_per_sample: number | null;
  audio_format: string | null;
  mime_type: string | null;
};

File: src-tauri/src/mapper/playlist/liked/track_liked_view.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct TrackLikedView {
    pub id: i64,
    pub liked_at: DateTime<Utc>, 
    pub path: String,
    pub library_cache_id: Option<i64>, 
    
    // --- Les champs issus de library_cache (lc) ---
    pub title: Option<String>,
    pub artist: Option<String>,
    pub album: Option<String>,
    pub duration: Option<f64>, // ou i64, selon comment tu as défini ta colonne en DB
    pub thumbnail_path: Option<String>,
    pub bits_per_sample: Option<u32>, // ou i64 selon la DB
    pub audio_format: Option<String>,
    pub mime_type: Option<String>,
}